pkg/storage/common.go

```diff
@@ -91,6 +91,16 @@ func ValidateManifest(imgStore ImageStore, repo, reference, mediaType string, bo
 				return "", zerr.ErrBadManifest
 			}
 		}
+
+		// a lightweight check if the blob is present
+		for _, layer := range manifest.Layers {
+			_, _, err := imgStore.CheckBlob(repo, layer.Digest)
```

Comment:
CheckBlob has a lock inside, and ValidateManifest is already locked from outside.

use imgStore.GetBlobContent() instead. it will throw ErrBlobNotFound.